internal/controller/sync_test.go

```diff
@@ -371,340 +367,634 @@ SOME GARBAGE
 	}
 }
 
-func TestRoute_hasNextPrivateKey(t *testing.T) {
-	// set up key for test cases
-	ecdsaKey, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
-	require.NoError(t, err)
-	ecdsaKeyPEM, err := utilpki.EncodePKCS8PrivateKey(ecdsaKey)
-	require.NoError(t, err)
+// trivial logic that re-implements OpenShift's IngressController behavior
```

Comment:
Reading this comment made me want to look at how things are implemented upstream. What about adding a quick link for anyone interested:

```suggestion
// Trivial logic that re-implements OpenShift's IngressController behavior. For context,
// the OpenShift IngressController code that deals with this is visible at:
// https://github.com/openshift/router/blob/72114ea/pkg/router/controller/status.go
```